### PR TITLE
rubocop: Autocorrect stanza order for `on_os` and `on_arch` blocks

### PR DESCRIPTION
**Pull request by issyl0**
This is a proof of concept for https://github.com/Homebrew/brew/pull/14976. That PR adds a RuboCop for ensuring a consistent order of `on_#{os_version}` and `on_#{arch}` blocks. In the review for that new feature it was deemed more consistent with current style to have the `on_#{os_version}` in order from oldest to newest.

What do you think?

-----

**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [ ] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [ ] `brew audit --cask --online <cask>` is error-free.
- [ ] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [ ] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [ ] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [ ] `brew audit --new-cask <cask>` worked successfully.
- [ ] `brew install --cask <cask>` worked successfully.
- [ ] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

File: Casks/deeper.rb [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
-  sha256 :no_check
-
   on_el_capitan do
     version "2.1.4"
     url "https://www.titanium-software.fr/download/1011/Deeper.dmg"
 
     depends_on macos: :el_capitan
   end
+
   on_sierra do
```
Comment (issyl0) [inline Casks/deeper.rb:9]: The audit as I've written it currently only takes into account _supported_ OS versions. Should I make it take into account ancient OS versions too (there's a list of those)?

**Comment by issyl0** [comment]
Hmmm by the looks of `brew readall` for casks this still needs some work for it to even be valid. Good to know. (Also I totally could have checked this myself without wasting CI time, had I remembered.)

I’ll revisit this tomorrow when I’m more awake (in theory).

**Comment by issyl0** [comment]
https://github.com/Homebrew/brew/pull/14976#issuecomment-1472939838

**Comment by reitermarkus** [comment]
Some cask seem to use a mix on `on_*` and top-level stanzas, e.g. top level `sha256` and override it in an `on_*` block, which will change semantics if the `on_*` is moved before the top-level `sha256`.

**Comment by reitermarkus** [comment]
Or do we evaluate `on_*` blocks late?

File: Casks/anki.rb [hunk 2]
```diff
@@ -5,16 +5,15 @@
...
-
   on_high_sierra :or_older do
     sha256 "9e9fe5ba51a3f6c43a568d15aaa4acb74992f013af17a0211b0447e0742588ec"
 
     url "https://github.com/ankitects/anki/releases/download/#{version}/anki-#{version}-mac-#{arch}-qt5.dmg",
         verified: "github.com/ankitects/anki/"
   end
 
+  url "https://github.com/ankitects/anki/releases/download/#{version}/anki-#{version}-mac-#{arch}-qt6.dmg",
+      verified: "github.com/ankitects/anki/"
```
Comment (reitermarkus) [inline Casks/anki.rb:16]: In my opinion, `url` and `sha256` should be in a `on_mojave :or_newer` block in this case.

**Comment by issyl0** [comment]
Closing this since it's easier to start again than resolve the conflicts! The changes in https://github.com/Homebrew/brew/pull/15013 should make these changes to stanza ordering fewer!

**Pull request closed without merging**